### PR TITLE
vmpool: improve test time

### PR DESCRIPTION
**Pull request by fossedihelm**
### What this PR does
This series of commits gradually remove complexity of pool tests to the point where we reduce the runtime from ~19 seconds down to ~250ms. On top of this we remove some unnecessary code and fix few issues.

### Special notes for your reviewer
These changes follow the logic introduced here https://github.com/kubevirt/kubevirt/pull/12468

### Release note

```release-note
NONE
```

### CODE REVIEW

**Comment by fossedihelm** [comment]
/hold
Based on https://github.com/kubevirt/kubevirt/pull/12505

**Review by 0xFelix — APPROVED** [review]
Would have been nice if the rename to `fakeVirtClient` could have been avoided but LGTM otherwise!

/lgtm

File: pkg/virt-controller/watch/pool_test.go [hunk 1]
```diff
@@ -650,10 +650,7 @@ var _ = Describe("Pool", func() {
 			nonMatchingVM.Name = fmt.Sprintf("%s-1", pool.Name)
 			nonMatchingVM.Labels = map[string]string{"madeup": "value"}
 			nonMatchingVM.OwnerReferences = []metav1.OwnerReference{}
-			vmSource.Add(nonMatchingVM)
-
-			// allow for cache to catch up. This non matching vm won't cause key to be queued
-			time.Sleep(1 * time.Second)
```
Comment (xpivarc) [inline pkg/virt-controller/watch/pool_test.go:658]: say goodbye to "flaky by nature"

File: pkg/virt-controller/watch/pool_test.go [hunk 2]
```diff
@@ -122,14 +120,11 @@ var _ = Describe("Pool", func() {
 		}
 
 		addVMI := func(vm *v1.VirtualMachineInstance, expectQueue bool) {
+			controller.vmiStore.Add(vm)
```
Comment (xpivarc) [inline pkg/virt-controller/watch/pool_test.go:110]: Can we remove the `expectQueue`?

**Review by xpivarc — COMMENTED** [review]
/approve
/hold
Feel free to unhold as my comment is more of nit

**Review by 0xFelix — APPROVED** [review]
/lgtm

**Review by 0xFelix — APPROVED** [review]
/lgtm

**Pull request merged**